f/common_logic/data_conversion.py

```diff
@@ -206,85 +209,47 @@ def json_to_csv(path: Path):
 @handle_file_errors
 def gpx_to_geojson(path: Path):
     """
-    Converts a GPX file to GeoJSON format.
-    Parses waypoints and track segments, returning a FeatureCollection.
+    Converts a GPX file to GeoJSON format using Fiona.
+    Reads all GPX layers (waypoints, tracks, etc.) and returns a FeatureCollection.
 
     Returns
     -------
     dict
         The converted GPX data as a GeoJSON dictionary.
     """
-    tree = ET.parse(path)
-    root = tree.getroot()
-    namespace = {"default": root.tag.split("}")[0].strip("{")}
-
     features = []
 
-    def parse_point(el):
-        lat = el.get("lat")
-        lon = el.get("lon")
-        props = {}
-        links = []
-
-        for child in el:
-            tag = child.tag.split("}")[-1]
-            if tag == "link":
-                href = child.attrib.get("href")
-                if href:
-                    links.append(href)
-                else:
-                    # Fallback: nested <text> tag for Locus variants
-                    text_el = child.find("./default:text", namespace)
-                    if text_el is not None and text_el.text:
-                        links.append(text_el.text.strip())
-                        logger.warning("Fallback to nested <text> tag for link")
-            else:
-                props[tag] = child.text
+    # Read all available layers in the GPX file
+    layers = fiona.listlayers(path)
 
-        if links:
-            props["link"] = ", ".join(links)
+    for layer in layers:
+        with fiona.open(path, layer=layer, driver="GPX") as collection:
+            for feature in collection:
+                properties = (
+                    dict(feature["properties"]) if feature["properties"] else {}
+                )
+
+                # Handle Locus Map link fields - combine all link-related fields
+                link_fields = [k for k in properties.keys() if "link" in k.lower()]
+                links = []
+                for field in sorted(link_fields):  # Sort for consistent output
+                    if properties[field]:
+                        links.append(str(properties[field]))
+                        del properties[field]
+
+                if links:
+                    properties["link"] = ", ".join(links)
+
+                # Clean up empty properties
```

Comment:
is this an artifact of reading GPX  (i.e. GPX or fiona adds empty properties that aren't real), or is this business logic?


I think this, and perhaps also the comma-joining links, are a lossy transformation that might be noted in the docstring